src/components/RecommendationPage.js

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import axiosInstance from '../apis/axiosClient';
+import { Typography, CircularProgress, List, ListItem } from '@mui/material';
+import { useAuth } from '@akosasante/react-auth-context';
+const RecommendationPage = () => {
+  const { user } = useAuth();
+  const userId = user ? user.userId : null;
+
+  const [recommendations, setRecommendations] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    if (userId) {
+      axiosInstance
+        .get(`/recommendations/${userId}`)
+        .then((response) => {
+          setRecommendations(response.data.recommendations);
+          setLoading(false);
+        })
+        .catch((error) => {
+          console.error('Error fetching recommendations:', error);
+          setError('An error occurred while fetching recommendations.');
+          setLoading(false);
+        });
+    } else {
+      setLoading(false);
+    }
+  }, [userId]);
+
+  return (
+    <div>
+      <Typography
+        fontWeight="bold"
+        color="text.secondary"
+        variant="h5"
+        spacing={0.5}
+        sx={{ mb: 2, p: 3 }}
+      >
+        Recommendations
+      </Typography>
+      {error ? (
+        <Typography color="error">{error}</Typography>
+      ) : loading ? (
+        <CircularProgress />
+      ) : (
+        <List>
+          {recommendations.map((album) => (
+            <ListItem key={album.albumName} sx={{ mb: 1 }}>
+              {album.albumName}
+            </ListItem>
+          ))}
+        </List>
+      )}
```

Comment:
In React, it's often a [code smell](https://martinfowler.com/bliki/CodeSmell.html) to have nested ternary operations in the render code (the code after `return` in a function component).

It is often preferred to have each state as it's own single line or statement using boolean operators instead. For example, the code here could be rewritten as:  
 

```suggestion
      {loading && <CircularProgress />}
      {error && <Typography color="error">{error}</Typography>}
      {!loading && !error && recommendations?.length === 0 && <Typography color="warning.main">Sorry no recommendations</Typography>}
      {!loading && !error && recommendations.length > 0 && (
        <List>
          {recommendations.map((album) => (
            <ListItem key={album.albumName} sx={{ mb: 1 }}>
              {album.albumName}
            </ListItem>
          ))}
        </List>
      )}
```

This works because the boolean operator `&&` checks each value to see if it's truthy, and if they are all truthy, it will return the value of the last item checked (in the above example, that would be the element we want to return in that case). 

This way when reading the code, the reader doesn't have to go through branches thinking "okay if error is false but loading is true then this element will be shown". Which can get tangled after many if/else/if/elses. Instead, each line can be read on its own: "if loading is true, then regardless of any other variables, the CircularProgress element will be shown". "if error is true, then the error message will be shown". "if error and loading are false, and the recommendations has an empty length, some text will be shown" etc.

Reply:
Akos I have done some changes previously in commit 0705dee8f06b5863550c25dd3f8460e13a43b671, should I change for your suggection better?

Reply:
I think what you have now is reasonable, no need to change it :+1: . Just keep in mind for future cases, this approach for avoiding nested ternaries in render code.